libraries/wallet/wallet.cpp

```diff
@@ -687,88 +687,108 @@ namespace golos { namespace wallet {
+                        auto req_posting_accs =_remote_database_api->get_accounts(std::vector<account_name_type>(
+                            req_posting_approvals.begin(), req_posting_approvals.end()));
+                        for (auto& acc : req_posting_accs) {
+                            approving_accounts[acc.name] = acc;
+                            fetch_keys(acc.posting);
+                            for (const auto& auth : acc.posting.account_auths) {
+                                posting_account_auths.push_back(auth.first);
+                            }
                         }
+                        total_accs_got += req_posting_accs.size();
                     }
 
-                    for( const account_name_type& acct_name : req_owner_approvals ) {
-                        const auto it = approving_account_lut.find( acct_name );
-                        if( it == approving_account_lut.end() )
-                            continue;
-                        const golos::api::account_api_object& acct = it->second;
-                        vector<public_key_type> v_approving_keys = acct.owner.get_keys();
-                        for( const public_key_type& approving_key : v_approving_keys ) {
-                            wdump((approving_key));
-                            approving_key_set.insert( approving_key );
+                    if (!active_account_auths.empty()) {
+                        auto active_account_auth_objs = _remote_database_api->get_accounts(active_account_auths);
+                        total_accs_got += active_account_auth_objs.size();
+                        for (auto& acc : active_account_auth_objs) {
+                            approving_accounts[acc.name] = acc;
+                            fetch_keys(acc.active);
+                        }
+                    }
+                    if (!owner_account_auths.empty()) {
+                        auto owner_account_auth_objs = _remote_database_api->get_accounts(owner_account_auths);
+                        total_accs_got += owner_account_auth_objs.size();
+                        for (auto& acc : owner_account_auth_objs) {
+                            approving_accounts[acc.name] = acc;
+                            fetch_keys(acc.owner);
                         }
                     }
+                    if (!posting_account_auths.empty()) {
+                        auto posting_account_auth_objs = _remote_database_api->get_accounts(posting_account_auths);
+                        total_accs_got += posting_account_auth_objs.size();
+                        for (auto& acc : posting_account_auth_objs) {
+                            approving_accounts[acc.name] = acc;
+                            fetch_keys(acc.posting);
+                        }
+                    }
+
+                    // checks if accounts for all names have got
+                    FC_ASSERT(approving_accounts.size() == total_accs_got,
```

Comment:
`total_accs_got` can contain wrong value, because transaction allowed to contain both active- and owner-signed operations. Move this assert after each `get_accounts`, it will fix this issue and make asserts more precise.

Reply:
@zxcat Do we need this assert at all? Why it need?
See above: "Testing improved FC_ASSERT on getting account from approving_account_lut"
Another assert fires. Not this one.

Reply:
@maslenitsa93, "lut" assert do almost the same check (actually, it should not fire on normal transactions, maybe on multisig). but if tx fails in "lut", it gives no point, what actually happened and which account missing. Having assert after each `get_acconts` gives much more precise information about fail.

Reply:
@zxcat I already improved lut-assert, it gives account name, see above again.
And size-assert will not give name, it just gives expected & actual size which is not informative, absolutely...

Reply:
@maslenitsa93, ok, you're right, adding name should be informative enough. you can just remove that assert and `total_accs_got` calculation.
In most cases it's better to fail early, but in this case it's not really necessary, and code will be simpler.